### PR TITLE
Logs panel: Support details view

### LINKED ISSUE #22271: Log panel: support details view
**What would you like to be added**:
It would be very useful if the Log panel had a details view toggle (as found in Explore). It would look and behave in exactly the same way (though I guess _without_ the +/- filter buttons)

**Why is this needed**:
We use Loki and our logs are in JSON. We find the explore details view very useful & intuitive for extra context. It would be great to have the same benefit directly accessible from the Log panel of a dashboard (this is quite common in other logging UIs e.g. when browsing AWS CloudWatch Events).

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:

This PR brings back logs details view in Logs Panel. We used to have `allowDetails` toggle which name was miss-leading cause `allowDetails === true` actually hided the details view. I have decided to keep this toggling props and add the option to hide log details in Logs panel. It defaults to `false`, so the log view is defaultly visible.

To display links, we are using `getFieldLinksForExplore` which I think wee should rename (but not sure how) as it is used outside of Explore, for example in TimeSeries panel to create links for exemplars. 

When we are using internal data source, it opens an Explore with that data source selected and specified query. If external link, it opens external link. 

The detected fields are filtering are not supported and options is not displayed. Filtering is not possible as we don't have data source in visualisation. For detected fields, we would need some UI work on how to show/toggle it. We can think about it if our users would find this useful. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/22271

**How to test this:**:
1. Run log query with logs panel and open detail vie in the same way as it is done in Explore. You should be able to see:
- Details (labels/detected fields)
- Derived fields with or without link, depending on your field. 
- Possibility to show stats
2. Check if everything stayed correct in Explore.

### CODE REVIEW

File: packages/grafana-ui/src/components/Logs/LogDetailsRow.test.tsx [hunk 1]
```diff
@@ -14,6 +14,9 @@ const setup = (propOverrides?: Partial<Props>) => {
     getStats: () => null,
```
Comment (Elfo404) [inline packages/grafana-ui/src/components/Logs/LogDetailsRow.test.tsx:14]: Just a nit: I know this test file is still using enzyme, would it make sense to write the new ones using testing-library instead?

**Review by Elfo404 — APPROVED** [review]
Code looks great! 🚀 🚀 

Regarding the panel option, I only find it confusing (but this is generally true, not only for this PR) to have "negating options" that require me to do some math to understand them.

In this case, I'd rather have "Enable log details" that defaults to `true`. What do you think?

File: packages/grafana-ui/src/components/Logs/LogDetailsRow.test.tsx (hunk 1 above)
Comment (ivanahuckova) [reply packages/grafana-ui/src/components/Logs/LogDetailsRow.test.tsx:13]: Yes, very good point. Done in https://github.com/grafana/grafana/pull/34125/commits/69063bb243ec262e20f7bedaabb45e5b04ac8b55

**Comment by ivanahuckova** [comment]
@Elfo404 yeah, good point. I have updated in https://github.com/grafana/grafana/pull/34125/commits/dad54da5e17f2d6cc3248a6964739b14e2524de8. I decided to update the naming everywhere so we are using the same names of variables and props for the same functionality.

**Review by Elfo404 — APPROVED** [review]
awesome! 🔥

**Pull request merged**